R/csv-utils.R

```diff
@@ -1,19 +1,42 @@
-#' @keywords internal
-mrgda_read_csv <- function(.file) {
+#' Read csv file
+#'
+#' @description
+#' Reads csv files and formats the data to replace any "." with
+#' NA values.
+#'
+#' @param .file file path to csv file
+#'
+#' @export
+read_nm_csv <- function(.file) {
```

Comment:
Consider letting the user pass `na.strings` through; it could default to `.`, but would let user pass additional `na.strings` as well. For example, `na.strings = c(".", "NA")`.

Reply:
In fact, why not add `...` to the function signature and let users pass anything through to `data.table::fread()`?.

Reply:
@kylebaron i'm definitely not opposed to that, but when `na.strings` is no longer `"."`, is it still an `nm` reader?

Reply:
Include argument `integer64 = "numeric"`  and pass to `fread()`.  This prevents `bit64::integer64` from getting into the data; this is what `readr` does by default. 

```r
> read_csv(I("a\n1000223424232"), show_col_types=FALSE)
# A tibble: 1 × 1                                                                                                          
              a
          <dbl>
1 1000223424232


> fread(text = "a\n1000223424232") %>% str
Classes ‘data.table’ and 'data.frame':	1 obs. of  1 variable:
 $ a:integer64 1000223424232 
 - attr(*, ".internal.selfref")=<externalptr> 

> fread(text = "a\n1000223424232", integer64 = "double") %>% str
Classes ‘data.table’ and 'data.frame':	1 obs. of  1 variable:
 $ a: num 1e+12
 - attr(*, ".internal.selfref")=<externalptr>  
```

Reply:
> @kylebaron i'm definitely not opposed to that, but when `na.strings` is no longer `"."`, is it still an `nm` reader?

Possibly; if not, then I'd remove `nm` from the name. Sometimes this stuff gets written into a data set during handoff etc and it'd be nice to keep using this function as a default option.